persist/storer_test.go

```diff
@@ -492,5 +474,60 @@ func TestFileStore_Save(t *testing.T) {
 
 	assert.Equal(t, "v", entry.Value)
 	assert.Equal(t, expectedTS, entry.Timestamp)
+}
+
+func TestFileStore_DeleteOldEntriesUponLoading(t *testing.T) {
+	// Reset global variable affected by other tests to the original
+	// value used by the library.
+	SetNow(time.Now)
+
+	// Given a file storer
+	filePath := path.Join(t.TempDir(), "test.json")
+	ttl := 1 * time.Second
 
+	storer, err := NewFileStore(filePath, log.NewStdErr(true), ttl)
+	assert.NoError(t, err)
+
+	// When a valid storer contains keys with timestamp greater than TTL
+	storer.Set("expiredKey", "val")
+
+	time.Sleep(time.Second * 2)
+	storer.Set("recentKey", "v")
+
+	assert.NoError(t, storer.Save())
+
+	storer, err = NewFileStore(filePath, log.NewStdErr(true), ttl)
+	assert.NoError(t, err)
+
+	time.Sleep(time.Second * 1)
```

Comment:
Why are we sleeping an extra second here, after adding the entry we do not expect to be expired?

Reply:
oh this is from some tests i was doing in order to debug an issue was caused because i didn't set up `SetNow(time.Now)` . I will remove it thanks! it should be not needed